File: app/backend/src/main/java/at/maturaballmanager/model/Address.java
package at.maturaballmanager.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;

import java.io.Serializable;

@Entity
public class Address extends PanacheEntity implements Serializable {

    @Column(length = 20, nullable = false)
    public String street;

    @Column(length = 10, nullable = false)
    public String door;

    @Column(length = 3)
    public String stage;

    @Column(length = 8, nullable = false)
    public String zipCode;

    @Column(length = 30, nullable = false)
    public String town;

    @Column(length = 30, nullable = false)
    public String country;

    public Address() {}
}

File: app/backend/src/main/kotlin/at/htlleonding/maturaballmanager/model/Address.kt
// src/main/kotlin/at/htlleonding/maturaballmanager/model/Address.kt

package at.htlleonding.maturaballmanager.model

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern

@Embeddable
data class Address(
    @NotBlank(message = "Straße ist erforderlich")
    var street: String? = null,

    @NotBlank(message = "Hausnummer ist erforderlich")
    @Column(name = "house_number")
    var houseNumber: String? = null,

    var floor: String? = null,

    var door: String? = null,

    @NotBlank(message = "Postleitzahl ist erforderlich")
    @Pattern(regexp = "^[0-9]{4,5}$", message = "Bitte eine gültige Postleitzahl eingeben")
    @Column(name = "postal_code")
    var postalCode: String? = null,

    @NotBlank(message = "Stadt ist erforderlich")
    var city: String? = null,

    var country: String? = "AT"
)